src/contracts/StakingChef.sol

```diff
@@ -0,0 +1,158 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "./Chef.sol";
+import "./MetricToken.sol";
+
+//TODO StakeMetric, updateStaker, removeStaker, updateAccumulatedStakingRewards, harvest and Claim can be moved to Chef
+
+contract StakingChef is Chef {
+    using SafeMath for uint256;
+    Staker[] private _stakes;
+    uint256 private _totalAllocPoint;
+    uint256 private _lifetimeShareValue;
+
+    MetricToken private _metric;
+
+    constructor(address metricTokenAddress) {
+        _metric = MetricToken(metricTokenAddress);
+        setMetricPerBlock(4);
+    }
+
+    mapping(address => bool) public override addressExistence;
+    // modifier nonDuplicated(msg.sender);
+
+// --------------------------------------------------------------------- staking functions
+    function stakeMetric(
+        address newAddress,
+        uint256 metricAmount,
+        uint256 newStartDate
+    ) external nonDuplicated(newAddress) {
+    if (areRewardsActive() && _totalAllocPoint > 0) {
+            updateAccumulatedStakingRewards();
+        }
+        Staker memory stake = Staker({
+            stakeAddress: newAddress,
+            metricAmount: metricAmount, 
+            startDate: newStartDate,
+            rewardDebt: metricAmount.mul(_lifetimeShareValue).div(ACC_METRIC_PRECISION),
+            claimable: 0
+        });
+
+        _stakes.push(stake);
+        _totalAllocPoint = _totalAllocPoint.add(stake.metricAmount);
+        _metric.transfer(address(this), stake.metricAmount);
```

Comment:
just realized _metric won't be transferred from the contract itself
`SafeERC20.safeTransferFrom(IERC20(metric), msg.sender, address(this), stake.metricAmount)`